src/librustc_typeck/check/_match.rs

```diff
@@ -867,6 +868,15 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             self.check_pat_walk(&field.pat, field_ty, def_bm, true);
         }
 
+        if tcx.sess.features.borrow().non_exhaustive {
+            // Require `..` if structure or enum has non_exhaustive attribute.
```

Comment:
Require `..` if structure or *variant* has the `non_exhaustive` attribute.
It looks like this patch doesn't implement `non_exhaustive` for variants yet, but `non_exhaustive` on `enum` should not restrict use of its variants.

Reply:
Adjusted this comment.